### ISSUE TITLE
oagbert model Killed

### ISSUE THREAD

**Issue by emanuelevivoli** [THUDM/cogdl#239]
## 🐛 Bug

I'm using oagbert for sentence embedding.

## To Reproduce

 I want to obtain the embedding of a list of sentences ( `len(corpus)=124`) so I use (as the example) the following code:
```python
tokenizer, bert_model = oagbert()
tokens = tokenizer(corpus, return_tensors="pt", padding=True)
batch_embeddings = bert_model(**tokens)
embeddings = embeddings[1]
```

## Expected behavior

embedding should be the a `torch.Size([124, 768])` but instead the `bert_model(**tokens)` says:
```bash
Killed
```

**Comment by cenyk1230** [comment]
Hi @emanuelevivoli,

The error seems to be caused by out-of-memory (OOM). What's the memory size of your server? Maybe you can try a smaller batch size (e.g., x=1).

**Comment by emanuelevivoli** [comment]
Hi @cenyk1230 ,
I tried changing the size to 2, so I decided to split the whole corpus in chunk of 50 with the following:
```python
tokenizer, bert_model = oagbert()

def divide_chunks(l, n):
  """ Yield successive n-sized 
  chunks from l.
  """
  for i in range(0, len(l), n):
      yield l[i:i + n]

embeddings = []
for i, batch_corpus in enumerate(divide_chunks(corpus, 2)):
    # `tokens` (shape): 3x torch.Size([2, 361])
    tokens = tokenizer(batch_corpus, return_tensors="pt",
                       padding=True)
    try:
        # bert_model output is composed by
        # `all_encoder_layers` (shape): torch.Size([2, 361, 768])
        # `pooled_output` (shape): torch.Size([2, 768])
        batch_embeddings = bert_model(**tokens)
        print(
            f"embedding done for chunk {i}, len {batch_embeddings[1].shape}")
    except Exception as e:
        print(f"bert_model stopped here by: {e}")

    # we take the `pooled_output`
    # one embedding vector of 768 for each sentence
    embeddings.extend(batch_embeddings[1])
```
even though the same error `Killed` occures:
```
embedding done for chunk 0, len torch.Size([2, 768])
...
embedding done for chunk 57, len torch.Size([2, 768])
embedding done for chunk 58, len torch.Size([2, 768])
embedding done for chunk 59, len torch.Size([2, 768])
Killed
```

I suppose the `Killed` error is not raised by `oagbert` but I still don't get why I get this error.
The command `free -h -t` shows:
```
(env) user@server:~/myproject$ free -h t
              total        used        free      shared  buff/cache   available
Mem:           62Gi       3,5Gi        58Gi       435Mi       1,0Gi        58Gi
Swap:         2,0Gi       2,0Gi       6,0Mi
```
So, I have some questions:
- Is it possible that I don't have enough Swap memory? 
- Is the `bert_model` ready to use for inference or do I need to set `bert_model.eval()` to avoid any gradient computation and memorization for updates and loss calculation?

Regards, 
Emanuele

**Comment by cenyk1230** [comment]
Hi @emanuelevivoli,

You can add `eval()` and `no_grad()` by yourself. For example, 

```python
tokenizer, bert_model = oagbert()
bert_model.eval()

def divide_chunks(l, n):
  """ Yield successive n-sized 
  chunks from l.
  """
  for i in range(0, len(l), n):
      yield l[i:i + n]

embeddings = []
for i, batch_corpus in enumerate(divide_chunks(corpus, 2)):
    # `tokens` (shape): 3x torch.Size([2, 361])
    tokens = tokenizer(batch_corpus, return_tensors="pt",
                       padding=True)
    try:
        # bert_model output is composed by
        # `all_encoder_layers` (shape): torch.Size([2, 361, 768])
        # `pooled_output` (shape): torch.Size([2, 768])
        with torch.no_grad():
            batch_embeddings = bert_model(**tokens)
        print(
            f"embedding done for chunk {i}, len {batch_embeddings[1].shape}")
    except Exception as e:
        print(f"bert_model stopped here by: {e}")

    # we take the `pooled_output`
    # one embedding vector of 768 for each sentence
    embeddings.extend(batch_embeddings[1])

```

**Comment by emanuelevivoli** [comment]
That's actually what I did, thanks
